src/audio/wav64.c

```diff
@@ -148,7 +151,20 @@ static inline void rsp_vadpcm_decompress(void *input, int16_t *output, bool ster
 
 #endif /* VADPCM_REFERENCE_DECODER */
 
-void raw_waveform_read(samplebuffer_t *sbuf, int base_rom_addr, int wpos, int wlen, int bps) {
+void raw_waveform_read(samplebuffer_t *sbuf, int current_fd, int wpos, int wlen, int bps) {
+	uint8_t* ram_addr = (uint8_t*)samplebuffer_append(sbuf, wlen);
+	int bytes = wlen << bps;
+
+	uint32_t t0 = TICKS_READ();
+	// Run the DMA transfer. We rely on libdragon's PI DMA function which works
+	// also for misaligned addresses and odd lengths.
+	// The mixer/samplebuffer guarantees that ROM/RAM addresses are always
+	// on the same 2-byte phase, as the only requirement of dma_read.
```

Comment:
This comment does not apply anymore. `read` will always work irrespective of the buffer alignment, thought it might force a memory copy if the alignment is not respected. I would just remove the comment.